### PR TITLE
Code Quality: Convert Multiple Localized Strings to ICU Message Format | Add custom number format

### PR DESCRIPTION
**Pull request by XTorLukas**
Resolved / Related Issues

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as Ready to build. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.

From #15503
TODO list: https://github.com/files-community/Files/issues/15503#issuecomment-2138404595

**Add custom number format**
Add better number format from default `1,000,000.000` to `1 000 000.000` via custom format for used this `{0, number}`

### CODE REVIEW

**Comment by XTorLukas** [comment]
For this 
image

image

**Comment by yaira2** [comment]
@XTorLukas which localization is that?

**Comment by XTorLukas** [comment]
> which localization is that?

This https://github.com/files-community/Files/pull/15514#issue-2326344833

File: src/Files.App/Extensions/MessageFormatExtensions.cs [hunk 1]
```diff
@@ -25,11 +25,32 @@ public static class MessageFormatExtensions
 		/// </summary>
 		private static readonly CultureInfo _locale = new(AppLanguageHelper.PreferredLanguage.Code);
 
+		/// <summary>
+		/// Custom value formatters for the message formatter.
```
Comment (0x5bfa) [inline src/Files.App/Extensions/MessageFormatExtensions.cs:29]:
```suggestion
		/// Gets custom value formatters for the message formatter.
```

File: src/Files.App/Extensions/MessageFormatExtensions.cs [hunk 2]
```diff
@@ -25,11 +25,32 @@ public static class MessageFormatExtensions
...
+			// Custom formatting for number values.
+			Number = (CultureInfo _, object? value, string? style, out string? formatted) =>
+			{
+				if (style is not null && style == string.Empty)
+				{
+					// Format the number '{0, number}'
+					formatted = string.Format($"{{0:#,##0}}", value);
+					return true;
+				}
+				formatted = null;
```
Comment (0x5bfa) [inline src/Files.App/Extensions/MessageFormatExtensions.cs:43]:
```suggestion

				formatted = null;
```

**Review by 0x5bfa — APPROVED** [review]
Otherwise LGTM

**Comment by yaira2** [comment]
When I test with English, the item count string isn't displayed (using this branch).

**Review by yaira2 — APPROVED** [review]

File: src/Files.App/Extensions/MessageFormatExtensions.cs (hunk 2 above)
Comment (XTorLukas) [reply src/Files.App/Extensions/MessageFormatExtensions.cs:44]: I have to finally set up a regular formatting using a keyboard shortcut

**Review by 0x5bfa — APPROVED** [review]

**Pull request merged**